go/cmd/dolt/commands/fetch.go

```diff
@@ -67,54 +68,69 @@ func (cmd FetchCmd) ArgParser() *argparser.ArgParser {
 	return cli.CreateFetchArgParser()
 }
 
+func (cmd FetchCmd) RequiresRepo() bool {
+	return false
+}
+
 // Exec executes the command
 func (cmd FetchCmd) Exec(ctx context.Context, commandStr string, args []string, dEnv *env.DoltEnv, cliCtx cli.CliContext) int {
 	ap := cli.CreateFetchArgParser()
 	help, usage := cli.HelpAndUsagePrinters(cli.CommandDocsForCommandString(commandStr, fetchDocs, ap))
 	apr := cli.ParseArgsOrDie(ap, args, help)
 
-	r, remainingArgs, err := env.RemoteForFetchArgs(apr.Args, dEnv.RepoStateReader())
+	queryist, sqlCtx, closeFunc, err := cliCtx.QueryEngine(ctx)
 	if err != nil {
-		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+		cli.PrintErrln(err)
+		return 1
 	}
-
-	validationErr := validateFetchArgs(apr, remainingArgs)
-	if validationErr != nil {
-		return HandleVErrAndExitCode(validationErr, usage)
+	if closeFunc != nil {
+		defer closeFunc()
 	}
 
-	refSpecs, err := env.ParseRefSpecs(remainingArgs, dEnv.RepoStateReader(), r)
+	query, err := constructInterpolatedDoltFetchQuery(apr)
 	if err != nil {
 		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
 	}
-
-	var verr errhand.VerboseError
-	dEnv.UserPassConfig, verr = getRemoteUserAndPassConfig(apr)
-	if verr != nil {
-		return HandleVErrAndExitCode(verr, usage)
-	}
-
-	srcDB, err := r.GetRemoteDBWithoutCaching(ctx, dEnv.DbData().Ddb.ValueReadWriter().Format(), dEnv)
+	_, _, err = queryist.Query(sqlCtx, query)
 	if err != nil {
 		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
 	}
 
-	prune := apr.Contains(cli.PruneFlag)
-	mode := ref.UpdateMode{Force: true, Prune: prune}
-	err = actions.FetchRefSpecs(ctx, dEnv.DbData(), srcDB, refSpecs, r, mode, buildProgStarter(downloadLanguage), stopProgFuncs)
-	if err != nil && err != doltdb.ErrUpToDate {
-		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
-	}
 	return HandleVErrAndExitCode(nil, usage)
 }
 
-// validateFetchArgs returns an error if the arguments provided aren't valid
-func validateFetchArgs(apr *argparser.ArgParseResults, refSpecArgs []string) errhand.VerboseError {
-	if len(refSpecArgs) > 0 && apr.Contains(cli.PruneFlag) {
-		// The current prune implementation assumes that we're processing all branch specs on the remote, so if an
-		// explicit ref spec is provided we refuse to execute
-		return errhand.BuildDError("--prune option cannot be provided with a ref spec").Build()
+// constructInterpolatedDoltFetchQuery constructs the sql query necessary to call the DOLT_FETCH() function.
+// Also interpolates this query to prevent sql injection.
+func constructInterpolatedDoltFetchQuery(apr *argparser.ArgParseResults) (string, error) {
+	var params []interface{}
+
+	var buffer bytes.Buffer
+	var first bool
+	first = true
+
+	buffer.WriteString("call dolt_fetch(")
+
+	writeToBuffer := func(s string) {
```

Comment:
I think this function is simple enough that you could just use a strings.join() for this.